v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraDMLGenerator.java

```diff
@@ -0,0 +1,455 @@
+    List<PreparedStatementValueObject<?>> values = new ArrayList<>();
+
+    for (Map.Entry<String, PreparedStatementValueObject<?>> entry : pkColumnNameValues.entrySet()) {
+      String colName = entry.getKey();
+      PreparedStatementValueObject<?> colValue = entry.getValue();
+      if (colValue.value() != null) {
+        allColumns.append(colName).append(", ");
+        placeholders.append("?, ");
+        values.add(colValue);
+      }
+    }
+
+    for (Map.Entry<String, PreparedStatementValueObject<?>> entry : columnNameValues.entrySet()) {
+      String colName = entry.getKey();
+      PreparedStatementValueObject<?> colValue = entry.getValue();
+      if (colValue.value() != CassandraTypeHandler.NullClass.INSTANCE) {
+        allColumns.append(colName).append(", ");
+        placeholders.append("?, ");
+        values.add(colValue);
+      }
+    }
+
+    if (allColumns.length() > 0) {
+      allColumns.setLength(allColumns.length() - 2);
+    }
+    if (placeholders.length() > 0) {
+      placeholders.setLength(placeholders.length() - 2);
+    }
+
+    String preparedStatement =
+        "INSERT INTO "
+            + tableName
+            + " ("
+            + allColumns
+            + ") VALUES ("
+            + placeholders
+            + ") USING TIMESTAMP ?;";
+
+    PreparedStatementValueObject<Long> timestampObj =
+        PreparedStatementValueObject.create("USING_TIMESTAMP", timestamp);
+    values.add(timestampObj);
+
+    return new PreparedStatementGeneratedResponse(preparedStatement, values);
+  }
+
+  /**
+   * Constructs a delete statement in CQL (Cassandra Query Language) using the provided table name,
+   * primary key values, and timestamp.
+   *
+   * @param tableName the name of the table from which records will be deleted.
+   * @param pkColumnNameValues a map containing the primary key column names and their corresponding
+   *     prepared statement value objects.
+   * @param timestamp the timestamp (in microseconds) to use for the delete operation.
+   * @return a {@link DMLGeneratorResponse} containing the generated CQL delete statement and a list
+   *     of values to bind to the prepared statement.
+   *     <p>This method: 1. Iterates through the provided primary key column values, appending
+   *     conditions to the WHERE clause of the CQL delete statement. 2. Constructs the `DELETE FROM`
+   *     CQL statement with the specified table name, primary key conditions, and a `USING
+   *     TIMESTAMP` clause. 3. Creates a list of values to be used with the prepared statement,
+   *     including the timestamp.
+   *     <p>If no primary key column values are provided, an empty WHERE clause is generated. An
+   *     exception may be thrown if any value type does not match the expected type.
+   */
+  private static DMLGeneratorResponse getDeleteStatementCQL(
+      String tableName,
+      Map<String, PreparedStatementValueObject<?>> pkColumnNameValues,
```

Comment:
Is there a strong reason that delete is not a prepared statement here?
Please consider using prepared statements to avoid sql injection issues.